README.md

```diff
@@ -1,39 +1,118 @@
+format. It's a solution to read and write ZIM file on many systems and
+architectures. More information about the ZIM format and the openZIM
+project at http://www.openzim.org/
 
-Use the zimlib in your own software --- for example, reader
-applications --- to make them ZIM-capable without the need having to
-dig too much into the ZIM file format.
+Disclaimer
+----------
 
-To build:
+This document assumes you have a little knowledge about software
+compilation. If you experience difficulties with the dependencies or
+with the ZIM libary compilation itself, we recommend to have a look to
+[kiwix-build](https://github.com/kiwix/kiwix-build).
+
+Preamble
+--------
+
+Although the ZIM library can be compiled/cross-compiled on/for many
+sytems, the following documentation explains how to do it on POSIX
+ones. It is primarly though for GNU/Linux systems and has been tested
+on recent releases of Ubuntu and Fedora.
+
+Dependencies
+------------
+
+The Kiwix library relies on many third parts software libraries. They
+are prerequisites to the Kiwix library compilation. Following
+libraries need to be available:
+
+* Z ................................................. http://zlib.net/
+(package zlib1g-dev on Ubuntu)
+* Bzip2 ......................................... http://www.bzip.org/
+(package libbz2-dev on Ubuntu)
+* Lzma ...................................... http://tukaani.org/lzma/
+(package lzma-dev on Ubuntu)
+* ICU ................................... http://site.icu-project.org/
+(package libicu-dev on Ubuntu)
+* Xapian ......................................... https://xapian.org/
+(package libxapian-dev on Ubuntu)
+
+These dependencies may or may not be packaged by your operating
+system. They may also be packaged but only in an older version. The
+compilation script will tell you if one of them is missing or too old.
+In the worse case, you will have to download and compile bleeding edge
+version by hand.
+
+If you want to install these dependencies locally, then use the ZIM
+library directory as install prefix.
+
+Environnement
+-------------
+
+The ZIM library builds using [Meson](http://mesonbuild.com/) version
+0.34 or higher. Meson relies itself on Ninja, pkg-config and few other
+compilation tools.
+
+Install first the few common compilation tools:
+* Automake
+* Libtool
+* Virtualenv
+* Pkg-config
+
+Then install Meson itself:
 ```
-./autogen.sh
-./configure
-make
+virtualenv -p python3 ./ # Create virtualenv
+source bin/activate      # Activate the virtualenv
+pip install meson        # Install Meson
+hash -r                  # Refresh bash paths
 ```
 
-OSX compilation
----------------
-On MacOSX, you'll need to install some packages from
-[homebrew](http://brew.sh/):
+Finally download and build Ninja locally:
 ```
-brew update
-brew install xz libmagic
+git clone git://github.com/ninja-build/ninja.git
+cd ninja
+git checkout release
+./configure.py --bootstrap
+mkdir ../bin
+cp ninja ../bin
```

Comment:
This 2 lines seem a fuzzy. What we need is to have the `ninja` binary in the `PATH` env variable.
This can be acheved in different ways :
- Add the ninja build directory to `PATH`
- Copy `ninja` binary into a directory already in `PATH`
- Copy `ninja` binary into a new directory and add this directory to `PATH`.

These 2 lines cannot ensure that the new `../bin` directory will be in the `PATH`.
I think it is preferable to simple say to the user what is needed (to have ninja binary in the PATH) and to provide a simple way to do it if he don't know how to do :
```
Put the `ninja` binary in a directory in the PATH environment variable. If you don't know how to do it, you can simply run :
`export PATH=$(pwd);$PATH`
```